### PR TITLE
Search

### PR DESCRIPTION
**Pull request by beetlejays**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -32,11 +37,39 @@ function App() {
...
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    setError(null);
+    if (search === '') {
+      return;
+    }
+
+    async function fetchSearchMovieData() {
+      const API_KEY = process.env.REACT_APP_API_KEY;
```
Comment (macfrei) [inline src/App.js:47]: You don't need this here as you already got your API key in line 11
```suggestion
```

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -1,9 +1,69 @@
-import "./Footer.css";
+import {Link} from 'react-router-dom';
+import styled from 'styled-components';
 
+import SearchIcon from '../../assets/search-blue.svg';
+/* import WatchlistIcon from '../../assets/watchlist-add.svg';
+ */
```
Comment (macfrei) [inline src/components/Footer/Footer.js:6]:
```suggestion
```

File: src/components/Footer/Footer.js [hunk 3]
```diff
@@ -1,9 +1,69 @@
...
 export default function Footer() {
   return (
-    <div className="footer">
-      <h2>This is the footer</h2>
-    </div>
+    <StyledFooter className="footer">
```
Comment (macfrei) [inline src/components/Footer/Footer.js:9]: Why use `className` when using styled-components?

File: src/components/Footer/Footer.js [hunk 4]
```diff
@@ -1,9 +1,69 @@
...
+            <FooterIcon src={SearchIcon} alt="" />
+            <FooterIconSearchText>Search</FooterIconSearchText>
+          </FooterIconContainer>
+        </Link>
+        {/*   <Link to="/watchlist">
+          <FooterIconContainer>
+            <FooterIcon src={WatchlistIcon} alt="" />
+            <FooterIconSearchText>Watchlist</FooterIconSearchText>
+          </FooterIconContainer>
+        </Link> */}
```
Comment (macfrei) [inline src/components/Footer/Footer.js:22]: Please remove unused code before merging into main.
```suggestion
```

File: src/components/Footer/Footer.spec.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import {render, screen} from '@testing-library/react';
+import {BrowserRouter} from 'react-router-dom';
+
+import Footer from './Footer';
+
+describe('Footer', () => {
+  it('renders footer', () => {
+    render(<Footer />, {wrapper: BrowserRouter});
+    const footerElement = screen.getByText('Search');
```
Comment (macfrei) [inline src/components/Footer/Footer.spec.js:9]: I think you should be able to query this by using `screen.getByRole('link', {name: 'Search})`. That would be a bit more precise and automatically tests the accessibility of your element.

**Review by macfrei — COMMENTED** [review]
Good job, added a few comments you might want to check out 👍

File: src/App.js (hunk 1 above)
Comment (beetlejays) [reply src/App.js:47]: I thought since const is block-scoped I have to declare the key separate?

File: src/components/Footer/Footer.js (hunk 2 above)
Comment (beetlejays) [reply src/components/Footer/Footer.js:6]: Worked on the watchlist page on the weekend while I was stuck. Will remove unused code and implement in the next user story.

File: src/App.js (hunk 1 above)
Comment (macfrei) [reply src/App.js:47]: No :)

Comment (macfrei) [reply src/App.js:47]: The `API_KEY` variable is declared in your `App` function, so within this function, the variable is _always_ reachable :)

File: src/components/Footer/Footer.js (hunk 3 above)
Comment (beetlejays) [reply src/components/Footer/Footer.js:9]: Old code, that was missed while deleting

File: src/components/Footer/Footer.js (hunk 4 above)
Comment (beetlejays) [reply src/components/Footer/Footer.js:22]: This is also is from the watchlist. Same mistake as above, stupid mistake

File: src/components/Footer/Footer.spec.js (hunk 5 above)
Comment (beetlejays) [reply src/components/Footer/Footer.spec.js:9]: Thanks! Will implement, still getting used to testing...

**Pull request merged**